src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -308,45 +309,73 @@ private static void CollectProperties(Dictionary<ObjectTypeProperty, CodeWriterD
             }
         }
 
-        private static void DeserializeIntoVariable(this CodeWriter writer, JsonSerialization serialization, Action<CodeWriter, CodeWriterDelegate> valueCallback, CodeWriterDelegate element)
+        private static void DeserializeIntoVariable(this CodeWriter writer, JsonSerialization serialization, Action<CodeWriter, CodeWriterDelegate> valueCallback,
+            CodeWriterDelegate element, bool? writeNullHandling = null)
         {
+            void WriteNullableScope(Action writeContent)
```

Comment:
It seems that every block in this method calls WriteNullableScope might be cleaner to add another methd like DeserializeIntoVaribleWithNullScope that would write the check and call into DeserializeIntoVarible.

Reply:
Done. However, personally, I don't think it is cleaner because it increases cyclomatic complexity since you have to check the serialization type twice now.